panel/backends/wayland/kwin_wayland/lxqtwmbackend_kwinwayland.cpp

```diff
@@ -0,0 +1,805 @@
+        {
+            if (window->windowState.testFlag(LXQtTaskBarPlasmaWindow::state::state_demands_attention))
+            {
+                if (!transientsDemandingAttention.values(leader).contains(window))
+                {
+                    transientsDemandingAttention.insert(leader, window);
+                    emit windowPropertyChanged(leader->getWindowId(), int(LXQtTaskBarWindowProperty::Urgency));
+                }
+            }
+            else if (transientsDemandingAttention.remove(window))
+            {
+                emit windowPropertyChanged(leader->getWindowId(), int(LXQtTaskBarWindowProperty::Urgency));
+            }
+        }
+        else
+        {
+            emit windowPropertyChanged(window->getWindowId(), int(LXQtTaskBarWindowProperty::Urgency));
+        }
+    });
+
+    connect(window, &LXQtTaskBarPlasmaWindow::skipTaskbarChanged, this, [window, this] {
+        updateWindowAcceptance(window);
+    });
+
+    // QObject::connect(window, &PlasmaWindow::applicationMenuChanged, q, [window, this] {
+    //     this->dataChanged(window, QList<int>{ApplicationMenuServiceName, ApplicationMenuObjectPath});
+    // });
+
+    // Handle transient.
+    if (LXQtTaskBarPlasmaWindow *leader = window->parentWindow.data())
+    {
+        transients.insert(window, leader);
+
+        // Update demands attention state for leader.
+        if (window->windowState.testFlag(LXQtTaskBarPlasmaWindow::state::state_demands_attention))
+        {
+            transientsDemandingAttention.insert(leader, window);
+            if(leader->acceptedInTaskBar)
+                emit windowPropertyChanged(leader->getWindowId(), int(LXQtTaskBarWindowProperty::Urgency));
+        }
+    }
+    else
+    {
+        windows.emplace_back(window);
+        updateWindowAcceptance(window);
+    }
+}
+
+bool LXQtWMBackend_KWinWayland::acceptWindow(LXQtTaskBarPlasmaWindow *window) const
+{
+    if(window->windowState.testFlag(LXQtTaskBarPlasmaWindow::state::state_skiptaskbar))
+        return false;
+
+    if(transients.contains(window))
+        return false;
+
+    return true;
+}
+
+void LXQtWMBackend_KWinWayland::updateWindowAcceptance(LXQtTaskBarPlasmaWindow *window)
+{
+    if(!window->acceptedInTaskBar && acceptWindow(window))
+    {
+        window->acceptedInTaskBar = true;
+        emit windowAdded(window->getWindowId());
+    }
+    else if(window->acceptedInTaskBar && !acceptWindow(window))
+    {
+        window->acceptedInTaskBar = false;
+        emit windowRemoved(window->getWindowId());
+    }
+}
+
+LXQtTaskBarPlasmaWindow *LXQtWMBackend_KWinWayland::getWindow(WId windowId) const
+{
+    for(auto &window : std::as_const(windows))
+    {
+        if(window->getWindowId() == windowId)
+            return window.get();
+    }
+
+    return nullptr;
+}
+
+static inline QByteArray detectDesktopEnvironment()
+{
+    const QByteArray xdgCurrentDesktop = qgetenv("XDG_CURRENT_DESKTOP");
```

Comment:
I have a concern, and a comment.

**Concern**: Will this work? When I was discussing with @stefonarch, he told me the following:

> startlxqtwayland script uses export XDG_CURRENT_DESKTOP="LXQt:$COMPOSITOR:wlroots" where $COMPOSITOR is the value set in session.conf, so kwin_wayland, sway, wayfire, labwc, ecc

This would mean that `XDG_CURRENT_DESKTOP` would be having a value `"LXQt:kwin_wayland"`, and we're not comparing against that.

**Comment**: If we change `getBackendScore()` to `getBackendScore( QString )` then this function would not be required at all.

Reply:
Let's say I split variable contents by `:` character. Then how do we use the resulting list of names?

Reply:
I've tried a loop which gives priority to first items in the list